### PR TITLE
Dact 561

### PR DESCRIPTION
**Pull request by rwalrondch**
Changes to validate nationality,  passing in a nationality list from env vars.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -10,7 +10,7 @@
         <relativePath/> <!-- lookup parent from repository -->
     </parent>
     <artifactId>officer-filing-api</artifactId>
-    <version>unversioned</version>
+    <version>poc-11-09-2023</version>
```
Comment (PhilipGlover) [inline pom.xml:13]: should this be included?

File: src/main/resources/application.properties
Comment (PhilipGlover) [inline src/main/resources/application.properties:1]: won't we need `NATIONALITY_LIST` in here too?   I've been off for a 2 weeks and I've forgotten the last 6 months, but..?

File: docker-compose.yml [hunk 2]
```diff
@@ -3,5 +3,23 @@ version: "3.8"
...
...Kuwaiti,Kyrgyz,Lao,Latvian,Lebanese,Liberian,Libyan,Liechtenstein citizen,Lithuanian,Luxembourger,Macanese,Macedonian,Malagasy,Malawian,Malaysian,Maldivian,Malian,Maltese,Marshallese,Martiniquais,Mauritanian,Mauritian,Mexican,Micronesian,Moldovan,Monegasque,Mongolian,Montenegrin,Montserratian,Moroccan,Mosotho,Mozambican,Namibian,Nauruan,Nepalese,New Zealander,Nicaraguan,Nigerian,Nigerien,Niuean,North Korean,Northern Irish,Norwegian,Omani,Pakistani,Palauan,Palestinian,Panamanian,Papua New Guinean,Paraguayan,Peruvian,Pitcairn Islander,Polish,Portuguese,Prydeinig,Puerto Rican,Qatari,Romanian,Russian,Rwandan,Salvadorean,Sammarinese,Samoan,Sao Tomean,Saudi Arabian,Scottish,Senegalese,Serbian,Citizen of Seychelles,Sierra Leonean,Singaporean,Slovak,Slovenian,Solomon Islander,Somali,South African,South Korean,South Sudanese,Spanish,Sri Lankan,St Helenian,St Lucian,Stateless,Sudanese,Surinamese,Swazi,Swedish,Swiss,Syrian,Taiwanese,Tajik,Tanzanian,Thai,Togolese,Tongan,Trinidadian,Tristanian,Tunisian,Turkish,Turkmen,Turks and Caicos Islander,Tuvaluan,Ugandan,Ukrainian,Uruguayan,Uzbek,Vatican citizen,Citizen of Vanuatu,Venezuelan,Vietnamese,Vincentian,Wallisian,Welsh,Yemeni,Zambian,Zimbabwean"
```
[the remark this answers is not in the document]
Comment (rwalrondch) [reply docker-compose.yml:22]: yes sorry i'll add it to Slack. htpps://github.com/companieshouse/chs-configs/pull/2461

File: src/main/resources/application.properties
Comment (rwalrondch) [reply src/main/resources/application.properties:1]: I didn't need it to work,  if there is something we would later need it for then I could add it(i did add it but removed it when it wasn't being used)

File: pom.xml (hunk 1 above)
Comment (rwalrondch) [reply pom.xml]: Pushed.

**Review by mdesaikns — COMMENTED** [review]
changes looks ok but I'd recommend turning off below option in intellij. Lot of changes are auto import mods made on the fly and are not intentional. 
image

**Review by AlisonGriffiths — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerAppointmentValidatorTest.java [hunk 3]
```diff
@@ -62,7 +62,10 @@ class OfficerAppointmentValidatorTest {
...
+        officerAppointmentValidator = new OfficerAppointmentValidator(logger, companyProfileService, apiEnumerations, allowedNationalities);
+        //added in some fake nationalities to be used in test length validation.
```
Comment (AlisonGriffiths) [inline src/test/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerAppointmentValidatorTest.java:67]: If not needed can the commented out code below be removed?

File: docker-compose.yml [hunk 4]
```diff
@@ -3,5 +3,23 @@ version: "3.8"
 services:
   officer-filing-api:
     image: '169942020521.dkr.ecr.eu-west-1.amazonaws.com/local/officer-filing-api:latest'
+    environment:
```
[the remark this answers is not in the document]
Comment (rwalrondch) [reply docker-compose.yml:6]: sorry yes, i added this when trying to get it working, wasn't needed but i forgot to remove.  removed again now.

File: src/test/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerAppointmentValidatorTest.java (hunk 3 above)
Comment (rwalrondch) [reply src/test/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerAppointmentValidatorTest.java:67]: Comments removed.

File: src/main/resources/application.properties
Comment (PhilipGlover) [reply src/main/resources/application.properties:1]: it'll be needed for the e2e environments I think.

File: docker-compose.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (PhilipGlover) [reply docker-compose.yml]: 👍 and I see it is approved

File: src/main/resources/application.properties
Comment (rwalrondch) [reply src/main/resources/application.properties:1]: Ok, i'll add it back in just in case.

**Review by mdesaikns — APPROVED** [review]

**Review by PhilipGlover — COMMENTED** [review]
The comments I've made have been addressed and I can approve them, though I note the import optimising that is mentioned above hasn't been done, and of course the conflicts that #125 has been merged in.

**Review by AlisonGriffiths — APPROVED** [review]

**Pull request merged**